packages/compiler/src/server/client-config-provider.ts

```diff
@@ -0,0 +1,78 @@
+import { Connection } from "vscode-languageserver/node.js";
+import { ServerHost } from "./types.js";
+
+/**
+ * TypeSpec client-side configuration provider
+ * Inspired by VS Code's WorkspaceConfiguration API for extensibility
+ */
+export interface ClientConfigProvider {
+  /**
+   * Get a configuration value by section key
+   * @param section The configuration section (e.g., "lsp.emit", "formatting.enabled")
+   * @param defaultValue Default value if not found
+   * @returns The configuration value or default
+   */
+  get<T>(section: string): T | undefined;
```

Comment:
Any reason to not have the known config itself be typed instead of having this be done at the point of usage?